cli/src/pcluster/config/cluster_config.py

```diff
@@ -2373,8 +2493,9 @@ def get_instance_types_data(self):
         result[instance_type_info.instance_type()] = instance_type_info.instance_type_data
         for queue in self.scheduling.queues:
             for compute_resource in queue.compute_resources:
-                instance_type_info = compute_resource.instance_type_info
-                result[instance_type_info.instance_type()] = instance_type_info.instance_type_data
+                for instance_type in compute_resource.instance_types:
+                    instance_type_info = SlurmFlexibleComputeResource.get_instance_type_info(instance_type)
```

Comment:
same consideration here, compute_resource should have a method to retrieve the instance_type_info_list